npm/rome/src/index.ts

```diff
@@ -0,0 +1,131 @@
+interface FormatFilesDebugOptions extends FormatFilesOptions {
+	/**
+	 * If `true`, you'll be able to inspect the IR of the formatter
+	 */
+	debug: boolean;
+}
+
+interface FormatContentDebugOptions extends FormatFilesOptions {
+	/**
+	 * If `true`, you'll be able to inspect the IR of the formatter
+	 */
+	debug: boolean;
+}
+
+interface FormatFilesOptions {
+	/**
+	 * Writes the new content to disk
+	 */
+	write: boolean;
+}
+
+interface FormatContentOptions {
+	/**
+	 * A virtual path of the file. You should add the extension,
+	 * so Rome knows how to parse the content
+	 */
+	filePath: string;
+	/**
+	 * The range where to format the content
+	 */
+	range: [number, number];
+}
+
+interface FormatResult {
+	// Not final
+	content: string;
+	// Not final
+	errors: string[];
+}
+
+interface FormatDebugResult {
+	// Not final
+	content: string;
+	// Not final
+	errors: string[];
+	// Available when in debug mode
+	ir: string | null;
+}
+
+interface ParseOptions {
+	/**
+	 * A virtual path of the file. You should add the extension,
+	 * so Rome knows how to parse the content
+	 */
+	filePath: string;
+}
+
+interface ParseResult {
+	/**
+	 * The CST of the code
+	 */
+	cst: string;
+	/**
+	 * The AST of the code
+	 */
+	ast: string;
+	// Not final
+	errors: string[];
+}
+
+function isFormatFilesDebug(options: any): options is FormatFilesDebugOptions {
```

Comment:
Is there a way to constrain `options to be a `FormatFilesOptions`? Something like

```suggestion
function isFormatFilesDebug(options: FormatFilesDebugOptions | FormatFilesOptions): options is FormatFilesDebugOptions {
```

Reply:
Not unfortunately. If we use this type signature, the compiler will compliant as soon as we try to inspect `.debug` because it doesn't exist in `FormatFilesOptions`

Reply:
I think in this case the TypeScript compiler wants you to write an explicit existence check for the property first:
```typescript
function isFormatFilesDebug(options: FormatFilesOptions): options is FormatFilesDebugOptions {
    return "debug" in options && options.debug === true;
}
```

Reply:
That's very ugly. Let's do it